middleware/logger.go

```diff
@@ -13,22 +13,43 @@ import (
 	"time"
 )
 
+// DefaultLogFormatter contains all the data needed to construct a log message
+type DefaultLogFormatter struct {
+	buf bytes.Buffer
+}
+
+// LogFormatter allows you to customize this middleware's output format.
+type LogFormatter interface {
+	FormatRequest(r *http.Request) LogFormatter
+	FormatResponse(status, bytes int, elapsed time.Duration)
+	Log()
+	Recover(err error)
```

Comment:
Question: would you prefer one dual-purpose method, or two methods, one called from Recoverer when there's a panic, and one from Logger during normal operation?